pkg/istio-agent/xds_proxy.go

```diff
@@ -572,3 +533,21 @@ func sendUpstreamWithTimeout(ctx context.Context, upstream discovery.AggregatedD
 		return err
 	}
 }
+
+// sendDownstreamWithTimeout sends discovery response with default send timeout.
+func sendDownstreamWithTimeout(downstream discovery.AggregatedDiscoveryService_StreamAggregatedResourcesServer,
```

Comment:
Does envoy have a timeout for send/recv on xds?

Reply:
I am not sure, let me have a check

Reply:
Normally i think for receive, it should be no.

Reply:
It seems envoy is using async client to handle xds stream.

```cpp
void GrpcMuxImpl::establishNewStream() {
  ENVOY_LOG(debug, "Establishing new gRPC bidi stream for {}", service_method_.DebugString());
  stream_ = async_client_->start(service_method_, *this);
  if (stream_ == nullptr) {
    ENVOY_LOG(warn, "Unable to establish new stream");
    handleFailure();
    return;
  }

  control_plane_stats_.connected_state_.set(1);
  for (const auto type_url : subscriptions_) {
    sendDiscoveryRequest(type_url);
  }
}
```

Reply:
I do not see Envoy setting timeout for XDS requests